src/jit/gtlist.h

```diff
@@ -186,8 +186,8 @@ GTNODE(SUB_LO           , "-Lo"          ,GenTreeOp          ,0,GTK_BINOP)
 GTNODE(SUB_HI           , "-Hi"          ,GenTreeOp          ,0,GTK_BINOP)
 GTNODE(DIV_HI           , "/Hi"          ,GenTreeOp          ,0,GTK_BINOP)
 GTNODE(MOD_HI           , "%Hi"          ,GenTreeOp          ,0,GTK_BINOP)
-GTNODE(MUL_LONG         , "*long"        ,GenTreeOp          ,1,GTK_BINOP) // A mul that returns the 2N bit result of an NxN multiply. This op
-                                                                           // is used for x86 multiplies that take two ints and return a long
+GTNODE(MUL_LONG         , "*long"        ,GenTreeMulLong     ,1,GTK_BINOP) // A mul that returns the 2N bit result of an NxN multiply. This op
+                                                                           // is used for multiplies that take two ints and return a long
```

Comment:
x86 doesn't need this new struct. I think you should put all the GenTreeMulLong stuff (here, gtstructs.h, gentree.h, etc.) under `#ifdef _TARGET_ARM_`.